Cargo.toml

```diff
@@ -22,5 +22,10 @@ criterion = "0.3"
 name = "decoding_benchmark"
 harness = false
 
+[[bench]]
+name = "large_image"
+harness = false
+
 [features]
-default = ["rayon"]
+default = ["rayon", "simd"]
+simd = []
```

Comment:
I reconsidered the style of the flag here, and maybe it shouldn't be a compile time flag. The arguments for a compile time flag are in general, size of codegen/compile time and compatibility. Neither of those should be an issue. The functions are monomorphized and using the intrinsics is valid for any `x86` platform—it's just that the function must stay unreachable on platforms that do not support _executing_ such intrinsics.

There are downsides to a `feature` flag in particular as well. It is not orthogonal between users. Indeed, if any dependent enables usage of `simd` then _all_ the dependent in the tree with compatible versions will get the feature (and thus run different code). It becomes more complex to include the library because anyone must now make a choice to expose this feature or not. `rayon` in particular had been made a feature to support `wasm` deployment in those versions where multi-threading was strict not possible and any mention of `rayon` lead to compilation failure.

Thus I would suggest making a _runtime_ feature selection and add a 'flag'  to the `Decoder` to toggle it on/off. The default may as well be on due to not having a _strict_ conformance test suite, but I'm open to input on that decision.

Reply:
So the main benefit I can see of a feature flag is: "if you disable the simd feature, there is no unsafe code at all" (maybe that can even be made explicit with a `#[forbid]`? Not 100% sure yet of what you can do for this)

As far as I'm aware, no other image format library (at least the ones I'm familiar with - libjxl, libjpeg-turbo and libpng up to a point) has a runtime flag to decide between SIMD or not, so there's that too...

Reply:
If that is the intent then it should be `#![cfg_attr(not(feature = "simd"), forbid(unsafe_code)]` (quite the mouthful).

Regarding the flag, I wouldn't call it `SIMD` but rather `platform_compatible`. Maybe even `strict`, referencing the attribute on glsl code. That is the intent is not the slow things down or speed them up but rather to make the result independent of platform—i.e. the decoding should be a pure function.

If we do _not_ have such a mode then this can effectively not be used in browsers, or it would allow to fingerprint the user's processor architecture quite clearly by decoding certain images to canvas.

Reply:
I'm not sure browsers care that much about that - I believe you can at least get the GPU *model* with JS. You can probably also do this in a couple of other ways, i.e. wasm-SIMD exists.

Anyway, renamed it to `platform_independent` (and flipped the logic). PTAL.

Reply:
Seems fine. In any case, at least Tor cares because when I try to get any `WEBGL_debug_renderer_info` then the result is null—thus no exact information on any hardware is available. I would really like if more browsers cared but alas that's the world when Browser vendors themselves _want_ to fingerprint their users.